--- a/swift/lib/AST/FunctionRefInfo.cpp
+++ b/swift/lib/AST/FunctionRefInfo.cpp
@@ -17,3 +17,4 @@
+#include <cstdint>
 #include "swift/AST/FunctionRefInfo.h"
 #include "swift/AST/DeclNameLoc.h"
 #include "swift/AST/Identifier.h"
--- a/swift-foundation/Sources/_FoundationCShims/include/_CStdlib.h
+++ b/swift-foundation/Sources/_FoundationCShims/include/_CStdlib.h
@@ -54,6 +54,2 @@
-#if __has_include(<math.h>)
-#include <math.h>
-#endif
-
 #if __has_include(<signal.h>)
 /// Guard against including `signal.h` on WASI. The `signal.h` header file
